lib/customer.rb

```diff
@@ -0,0 +1,41 @@
+require 'csv'
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address
+
+  def initialize (id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all
+    customer_csv = CSV.read("data/customers.csv")
+    all_customers = []
+
+    customer_csv.each do |info|
+      id = info[0].to_i
+      email = info[1]
+      address = { 
+        :street => info[2],
+        :city => info[3],
+        :state => info[4],
+        :zip => info[5]
+      }
+    
+      new_customer = self.new(id, email, address)
+      all_customers << new_customer
+    end 
+    
+    return all_customers
+  end
+
+  def self.find(id)
+    desired_customer = (self.all).find { |order| order.id == id }
+    return desired_customer
+
+    desired_customer.empty? ? nil : desired_customer
```

Comment:
This is "dead code".   Line 38 can't ever run because of the `return` on line 36.

```
/Users/kaida/Developer/ada/cohort-13/grocery-store/lib/customer.rb:38: warning: statement not reached
```